api/src/client/dynamics/housing/DynamicsHousingRegistrationClient.ts

```diff
@@ -1,31 +1,45 @@
 import {
   getPropertyInterestTypeIntegerFromTitle,
-  HotelMotelRegistrationClient,
+  HousingRegistrationClient,
   HousingRegistrationRequest,
   HousingRegistrationRequestResponse,
   HousingRegistrationStatus,
 } from "@client/dynamics/housing/types";
 import { LogWriterType } from "@libs/logWriter";
 import { parseDate } from "@shared/dateHelpers";
+import { PropertyInterestType } from "@shared/housing";
 import axios, { AxiosError } from "axios";
 
-export const DynamicsHotelMotelRegistrationClient = (
+export const DynamicsHousingRegistrationClient = (
   logWriter: LogWriterType,
   orgUrl: string
-): HotelMotelRegistrationClient => {
-  const getHotelMotelRegistration = async (
+): HousingRegistrationClient => {
+  const getHousingRegistration = async (
     accessToken: string,
     propertyInterestId: string,
-    buildingCount: number
+    buildingCount: number,
+    propertyInterestType: PropertyInterestType
   ): Promise<HousingRegistrationRequestResponse> => {
     const logId = logWriter.GetId();
     logWriter.LogInfo(`Dynamics Hotel Motel Registration Client - Id:${logId}`);
-    const hotelPropertyInterestType = getPropertyInterestTypeIntegerFromTitle("Hotel");
-    const motelPropertyInterestType = getPropertyInterestTypeIntegerFromTitle("Motel");
+    let propertyInterestTypeFilter = "";
+
+    switch (propertyInterestType) {
```

Comment:
Is there a particular reason you reached for a `switch` here? I'm assuming we're anticipating more `propertyInterestType`s will be added in the future?

Reply:
Yup, for expansion potential. There's a lot of types here